src/ado2gh/Commands/ShareServiceConnectionCommand.cs

```diff
@@ -65,10 +65,17 @@ public async Task Invoke(string adoOrg, string adoTeamProject, string serviceCon
             var ado = _adoApiFactory.Create(adoPat);
 
             var adoTeamProjectId = await ado.GetTeamProjectId(adoOrg, adoTeamProject);
-            // TODO: If the service connection is already shared with this team project this will crash
-            await ado.ShareServiceConnection(adoOrg, adoTeamProject, adoTeamProjectId, serviceConnectionId);
 
-            _log.LogSuccess("Successfully shared service connection");
+            if (await ado.ContainsServiceConnection(adoOrg, adoTeamProject, adoTeamProjectId, serviceConnectionId))
+            {
+                _log.LogInformation("Service connection already shared with team project");
+            }
+            else
+            {
+                await ado.ShareServiceConnection(adoOrg, adoTeamProject, adoTeamProjectId, serviceConnectionId);
+
+                _log.LogSuccess("Successfully shared service connection");
+            }
```

Comment:
nit: To prevent extra indentation I would re-write this to the following:
```suggestion
            if (await ado.ContainsServiceConnection(adoOrg, adoTeamProject, adoTeamProjectId, serviceConnectionId))
            {
                _log.LogInformation("Service connection already shared with team project");
                return;
            }

            await ado.ShareServiceConnection(adoOrg, adoTeamProject, adoTeamProjectId, serviceConnectionId);

            _log.LogSuccess("Successfully shared service connection");
```